pandas/tests/dtypes/test_common.py

```diff
@@ -392,6 +392,51 @@ def test_is_not_unsigned_integer_dtype(dtype):
     assert not com.is_unsigned_integer_dtype(dtype)
 
 
+@pytest.mark.parametrize(
+    "dtype",
+    [
+        "int_",
+        "int8",
+        "Int8",
+        "int16",
+        "Int16",
+        "int32",
+        "Int32",
+        "int64",
+        "Int64",
+        "uint8",
+        "uint16",
+        "uint32",
+        "uint64",
+        "float_",
+        "float16",
+        "float32",
+        "Float32",
+        "float64",
+        "Float64",
+    ],
+)
+def test_all_type_comparison_with_true_result(dtype):
+    # GH#43038
+    assert pandas_dtype(dtype) == dtype
+
+
+@pytest.mark.parametrize(
+    "dtype,expected",
+    [
+        ("Int8", "int8"),
```

Comment:
you can use the fixture here as well

Reply:
@pytest.fixture
def dtype():
 return [
        "int_",
        "int8",
        "Int8",
        "int16",
        "Int16",
        "int32",
        "Int32",
        "int64",
        "Int64",
        "uint8",
        "uint16",
        "uint32",
        "uint64",
        "float_",
        "float16",
        "float32",
        "Float32",
        "float64",
        "Float64",
    ]
def test_all_type_comparison_with_true_result(dtype):
    # GH#43038
    for element in dtype:
        assert pandas_dtype(element) == element



@pytest.fixture
def dtype_and_expected():
 return [ 
        ("Int8", "int8"),
        ("Int16", "int16"),
        ("Int32", "int32"),
        ("Int64", "int64"),
        ("Float32", "float32"),
        ("Float64", "float64"),
    ]
def test_all_type_comparison_with_false_result(dtype_and_expected):
    # GH#43038
    for element in dtype_and_expected:
        assert not pandas_dtype(element[0]) == element[1]

Reply:
@jreback is this the change you want?

Reply:
use the existing fixtures do not create new ones

Reply:
is there an example that uses these fixtures you say?

Reply:
def test_all_type_comparison_with_true_result(any_real_numpy_dtype):
    # GH#43038
    assert pandas_dtype(any_real_numpy_dtype) == any_real_numpy_dtype

Reply:
@jreback is this the thing i need to do?

Reply:
yes